sdk/servicebus/service-bus/src/session/messageSession.ts

```diff
@@ -1101,9 +1018,9 @@ export class MessageSession extends LinkEntity {
       };
 
       if (this.isOpen()) {
+        addCreditAndSetTimer(true);
         this._receiver!.on(ReceiverEvents.message, onReceiveMessage);
         this._receiver!.on(ReceiverEvents.receiverDrained, onReceiveDrain);
-        addCreditAndSetTimer(true);
```

Comment:
Does the ordering matter?

Reply:
It shoudnt...

Reply:
It shouldn't - there's no async code so no messages can be received before (that I can tell).

This ordering is what we actually have in the non-session BatchingReceiver so I figured it was okay.

Reply:
Cool!